src/message.coffee

```diff
@@ -21,15 +21,15 @@ class Message extends EventEmitter
 
     # Keep track of when this message actually times out.
     @timedOut = false
-    do trackTimeout = =>
+    do trackTimeoutFunction = =>
       return if @hasResponded
 
       soft = @timeUntilTimeout()
       hard = @timeUntilTimeout true
 
       # Both values have to be not null otherwise we've timedout.
       @timedOut = not soft or not hard
-      setTimeout trackTimeout, Math.min soft, hard unless @timedOut
+      @trackTimeout = setTimeout trackTimeoutFunction, Math.min soft, hard unless @timedOut
```

Comment:
How about
```coffee
unless @timedOut
  clearTimeout @trackTimeoutId # Note you can pass null/undefined without issue
  @trackTimeoutId = setTimeout ...
```
This way you are guaranteed to always clear the timeout id before creating a new one and the logic stays in this one function.